mat64/triangular.go

```diff
@@ -395,6 +396,76 @@ func (t *TriDense) MulTri(a, b Triangular) {
 	}
 }
 
+// Exp calculates the exponential of the matrix a, e^a, placing the result
+// in the receiver. If a has an odd number of rows, the matrix will be converted
+// to a Dense to perform the calculation.
+func (t *TriDense) Exp(a Triangular) {
+	n, kind := a.Triangle()
+	t.reuseAs(a.Triangle())
```

Comment:
This needs a check that the kind of `t` is the same as `a`. 

@kortschak maybe reuseAs should check that the kind matches the receiver kind if isZero returns false? This is nice looking code.